// jest-mock-axios/dist/cancel/CancelToken.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _Cancel = require("./Cancel");

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * Based of https://github.com/axios/axios
 * Copyright (c) 2014-present Matt Zabriskie (MIT License)
 */
var CancelToken = /*#__PURE__*/function () {
  _createClass(CancelToken, null, [{
    key: "source",
    value: function source() {
      var cancel;
      var token = new CancelToken(function (c) {
        return cancel = c;
      });
      return {
        cancel: cancel,
        token: token
      };
    }
  }]);

  function CancelToken(executor) {
    _classCallCheck(this, CancelToken);

    var token = this;
    this.promise = new Promise(function (resolve) {
      token.resolvePromise = resolve;
    });
    executor(function cancel(message) {
      if (token.reason || !token.resolvePromise) {
        return;
      }

      token.reason = new _Cancel.Cancel(message);
      token.resolvePromise(token.reason);
    });
  }

  _createClass(CancelToken, [{
    key: "throwIfRequested",
    value: function throwIfRequested() {
      if (this.reason) {
        throw this.reason;
      }
    }
  }]);

  return CancelToken;
}();

exports["default"] = CancelToken;
